base/sparse/sparsematrix.jl

```diff
@@ -1447,6 +1447,8 @@ sparse(S::UniformScaling, m::Integer, n::Integer=m) = speye_scaled(S.λ, m, n)
 conj!(A::SparseMatrixCSC) = (@inbounds broadcast!(conj, A.nzval, A.nzval); A)
 (-)(A::SparseMatrixCSC) = SparseMatrixCSC(A.m, A.n, copy(A.colptr), copy(A.rowval), map(-, A.nzval))
 
+# the rest of real, conj, imag are handled correctly via AbstractArray methods
+imag(A::SparseMatrixCSC{Tv,Ti}) where {Tv<:Number,Ti} = spzeros(Tv, Ti, A.m, A.n)
```

Comment:
https://github.com/JuliaLang/julia/blob/7a05f686a27d9e070efb523b51420f90d9310e43/base/sparse/sparsevector.jl#L1025
Seems like there has been some inconsistency here for `AbstractArray`/`SparseMatrixCSC`/`SparseVector`.